MIGRATION.md

```diff
@@ -102,7 +102,113 @@ const storeEndpoint = api.getStoreEndpointWithUrlEncodedAuth();
+If you rely on a custom transport, you will need to make some adjustments to how it is created when migrating
+to v7. Note that we changed our transports from a class-based to a functional approach, meaning that
+the previously class-based transports are now created via functions. This also means that custom transports
+are now passed by specifying a factory function in the `Sentry.init` options object instead passing the custom
+transport's class.
+
+The following example shows how to create a custom transport in v7 vs. how it was done in v6:
+
+```js
+// New in v7:
+import { BaseTransportOptions, Transport, TransportMakeRequestResponse, TransportRequest } from '@sentry/types';
+import { createTransport } from '@sentry/core';
+
+export function makeMyCustomTransport(options: BaseTransportOptions): Transport {
+  function makeRequest(request: TransportRequest): PromiseLike<TransportMakeRequestResponse> {
+    // this is where your sending logic goes
+    const myCustomRequest = {
+      request.body,
+      url: options.url
+    };
+    return sendMyCustomRequest(myCustomRequest).then(response => ({
+      headers: {
+        'x-sentry-rate-limits': response.headers.get('X-Sentry-Rate-Limits'),
+        'retry-after': response.headers.get('Retry-After'),
+      },
+    }));
+  }
+
+  // `createTransport` takes care of rate limiting and flushing
+  return createTransport({ bufferSize: options.bufferSize }, makeRequest);
+}
+
+Sentry.init({
+  dsn: '...',
+  transport: makeMyCustomTransport, // this function will be called when the client is initialized
+  ...
+})
+
+
+
+// Before:
+class MyCustomTransport extends BaseTransport {
+  constructor(options: TransportOptions) {
+    // initialize your transport here
+    super(options);
+  }
+
+  public sendEvent(event: Event): PromiseLike<Response> {
+    // this is where your sending logic goes
+    // `url` is decoded from dsn in BaseTransport
+    const myCustomRequest = createMyCustomRequestFromEvent(event, this.url);
+    return sendMyCustomRequest(myCustomRequest).then(() => resolve({status: 'success'}));
+  }
+
+  public sendSession(session: Session): PromiseLike<Response> {...}
+  // ...
+}
+
+Sentry.init({
+  dsn: '...',
+  transport: MyCustomTransport, // the constructor was called when the client was initialized
+  ...
+})
+```
+
+Overall, the new way of transport creation allows you to create your custom sending implementation
+without having to deal with the conversion of events or sessions to envelopes.
+We recommend calling `createTransport` as demonstrated in the example above which, besides creating the `Transport`
+object with your custom logic, will also take care of rate limiting and flushing.
+
+Passing in the factory function instead of an initialized transport provides the advantage that you do not have to take
+care of decoding the URL from your DSN. Essentially, it provides the same abstraction level as in v6
+where you passed in the class instead of an instance.
```

Comment:
Don't think we need this here - but will leave it for you to make the final call.

Reply:
Good point, I agree that it's not necessary - will remove it.